resources/file.go

```diff
@@ -443,6 +443,46 @@ func (obj *FileRes) fileCheckApply(apply bool, src io.ReadSeeker, dst string, sh
 	return sha256sum, false, dstFile.Sync()
 }
 
+// dirCheckApply is the CheckApply operation for an empty directory
+func (obj *FileRes) dirCheckApply(apply bool) (bool, error) {
+	// Check if the path exists and is a directory
+	st, err := os.Stat(obj.path)
+	if err != nil && !os.IsNotExist(err) {
+		return false, err
+	}
+
+	if err == nil && st.IsDir() {
+		return true, nil // Already a directory, nothing to do
+	}
+	if err == nil && !st.IsDir() && !obj.Force {
+		return false, fmt.Errorf("Can't force file into dir: %s", obj.path)
+	}
+
+	if !apply {
+		return false, nil
+	}
+
+	// The path exists and is not a directory
+	// Delete the file if force is given
+	if err == nil && !st.IsDir() {
+		log.Printf("dirCheckApply: Removing (force): %s", obj.path)
+		if err := os.Remove(obj.path); err != nil {
+			return false, err
+		}
+	}
+
+	// Create the empty directory
+	mode := os.ModePerm
+	if obj.Mode != "" {
+		mode, err = obj.mode()
+		if err != nil {
+			return false, err
+		}
+	}
+	err = os.MkdirAll(obj.path, mode)
```

Comment:
When creating a dir, think of it as Force allows us to MkdirAll where as without force we're only allowed to Mkdir (no automatic parents).

This is safer, because you want to explicitly say you want parents or not.